framework/src/modules/http_api/controllers/node.js

```diff
@@ -287,6 +287,12 @@ async function _getForgingStatus(publicKey) {
 	const forgersPublicKeys = {};
 
 	Object.keys(keyPairs).forEach(key => {
+		// Convert publicKey to buffer when received as object (ie.: { type: 'Buffer', data: [] })
```

Comment:
@lsilvs What can cause it to be received as an object with `type` of `Buffer`?
I would be better to always deal with strings if possible.

Reply:
@jondubois Our RPC implementation convert the data before sending it back by doing `Buffer.from(JSON.stringify(message))`.
The `JSON.stringify` convert buffers to `{ type: 'Buffer', data: [] }`. That's why I had to convert it back here.

Ideally we should always transport strings but that is the current behaviour of `chain:getForgersKeyPairs` action.
This action will be removed as part of https://github.com/LiskHQ/lisk/issues/3033 anyway

Reply:
My view is that the complexity of converting to a Buffer is not worth it to begin with. The socket (Unix socket, TCP socket or whatever) will convert strings to binary anyway before sending them and it will probably do it at a lower level in a more efficient way.

This is similar to something we did in the past with block serialisation and tests showed that there was no benefit in that case so we removed it.

See https://github.com/LiskHQ/lisk/issues/1621 and https://github.com/LiskHQ/lisk/pull/1683

Reply:
This conversion is done by https://github.com/tj/axon through https://github.com/tj/node-amp-message. Unless we decide not to use Axon, I would keep this behaviour.

Reply:
The solution is that we should avoid sending Buffers through the axon socket to begin with because that complicates the communication protocol between modules but I guess if it is the current behaviour and too time consuming to fix, we can do it later (as the TODO comment indicates).